pkg/plugins/policies/meshhttproute/xds/filters.go

```diff
@@ -68,3 +71,37 @@ func responseHeaderModifier(mod api.HeaderModifier, envoyRoute *envoy_route.Rout
 	envoyRoute.ResponseHeadersToAdd = append(envoyRoute.ResponseHeadersToAdd, options...)
 	envoyRoute.ResponseHeadersToRemove = append(envoyRoute.ResponseHeadersToRemove, removes...)
 }
+
+func requestRedirect(redirect api.RequestRedirect, envoyRoute *envoy_route.Route) {
+	envoyRedirect := &envoy_route.RedirectAction{}
+	if redirect.Hostname != nil {
+		envoyRedirect.HostRedirect = string(*redirect.Hostname)
+	}
+	if redirect.Port != nil {
+		envoyRedirect.PortRedirect = uint32(*redirect.Port)
+	}
+	if redirect.Scheme != nil {
+		envoyRedirect.SchemeRewriteSpecifier = &envoy_route.RedirectAction_SchemeRedirect{
+			SchemeRedirect: *redirect.Scheme,
+		}
+	}
+
+	switch pointer.DerefOr(redirect.StatusCode, 301) {
+	case 301:
+		envoyRedirect.ResponseCode = envoy_route.RedirectAction_MOVED_PERMANENTLY
+	case 302:
+		envoyRedirect.ResponseCode = envoy_route.RedirectAction_FOUND
+	case 303:
+		envoyRedirect.ResponseCode = envoy_route.RedirectAction_SEE_OTHER
+	case 307:
+		envoyRedirect.ResponseCode = envoy_route.RedirectAction_TEMPORARY_REDIRECT
+	case 308:
+		envoyRedirect.ResponseCode = envoy_route.RedirectAction_PERMANENT_REDIRECT
+	default:
+		panic("impossible redirect")
```

Comment:
nitpick: same as in previous PR

Reply:
marking this resolved per the discussion